File: app/src/main/java/com/jcr/sharedtasks/util/DeepLinkUtils.java
package com.jcr.sharedtasks.util;

import android.net.Uri;

import com.jcr.sharedtasks.model.ProjectReference;

public final class DeepLinkUtils {

    public static final String DEEPLINK_HOST = "https://com.jcr.sharedtasks/";

    public static ProjectReference parseProjectUUID(Uri data) {
        String deepLinkPath = data.getEncodedPath().toString();
        String[] splitDeepLink = deepLinkPath.split("/");
        String projectName = splitDeepLink[1];
        String projectUUID = splitDeepLink[2];
        return new ProjectReference(projectUUID, projectName.replace("&", " "));
    }
}

File: app/src/main/java/com/jcr/sharedtasks/util/DeepLinkUtils.kt
package com.jcr.sharedtasks.util

import android.net.Uri

import com.jcr.sharedtasks.model.ProjectReference

object DeepLinkUtils {

    val DEEPLINK_HOST = "https://com.jcr.sharedtasks/"

    fun parseProjectUUID(data: Uri): ProjectReference {
        val deepLinkPath = data.encodedPath?.toString()
        val splitDeepLink = deepLinkPath?.split("/".toRegex())?.dropLastWhile { it.isEmpty() }?.toTypedArray()
        val projectName = splitDeepLink?.get(1)
        val projectUUID = splitDeepLink?.get(2)
        return ProjectReference(projectUUID, projectName?.replace("&", " "))
    }
}
